content/docs/reference/public_private_keys/index.md

```diff
@@ -0,0 +1,97 @@
+    Definition of public and private keys
+
+draft: false # Change this to "true" to hide the page
+toc: false # Change this to "true" to show the table of contents
+weight: 205 # For single pages, lower is first.
+---
+
+## Definition
+A private key is used in asymmetric key cryptography. Asymmetric key cryptography is based on applying mathematical functions to numbers to achieve personal secrecy. It uses two keys, one being the private key. If you think of decryption as locking and unlocking padlocks with keys, then the padlock that is locked with a public key can only be unlocked with its corresponding private key.
+
+On the other hand, public keys are distributed to the trusted masses. This is done through a public-key distribution channel. This channel should provide authentication and integrity. Someone should not send their public key to the community pretending to have a different public key. Everyone should have their own private and public keys. For example, Colin only needs one private key to receive all correspondence in the community, but Barbara needs *n* public keys to communicate with *n* entities in the community, one public key for each entity. In other words, Barbara needs a ring of public keys.
+
+## How we do it
+
+A key in the @protocol can be formed by using any alphanumeric and special characters (UTF-8) excluding "@", ":" and a white space (" "). A key in a secondary can be any of the following 5 types:
+
+1. Public Key
+
+    - A public key is a key which can be looked up by any @sign user.
+
+    - A public key should be part of the *scan* verb result.
+
+    - Format of the public key should be **public**:\<identifier>:**<@sign>**
+
+    **Example:**    
+
+    ```public:location@alice```
+
+    > The owner of the secondary should be allowed to update or delete the value of a public key.
+
+
+2. Private Key
+        
+    - A private key is a key which cannot be looked up any @sign user other than the one created it.
+    
+    - A private key should not be returned in a *scan* verb result.
+    
+    - Format of the private key should be **privatekey**:\<identifier>:**<@sign>**
+
+    **Example:**    
+
+    ```privatekey:pk1@alice```
+
+    > The owner of the secondary should be allowed to update or delete the value of a private key.
+
+
+3. User key
+    - A user key can only be looked up by an @sign user with whom the data has been shared.
+    - A user key should be part of the *scan* verb result only for the user who created it and the specific user it has been shared with.
+    - Format of the key shared with someone else should be   
+    **cached**:**<Shared with @sign>**:\<identifier>:**<Created by @sign>**
+
+    **Example:**    
+
+    ```@bob:phone@alice```
+    
+    > Note: Above Key should be part of scan verb result for only @alice and @bob 
+
+    > The owner of the secondary should be allowed to update or delete the value of a user key.
+ 
+
+4. Internal Key
```

Comment:
Please change this to be called [Private Key](https://github.com/atsign-foundation/at_tools/blob/trunk/at_commons/lib/src/keystore/at_key.dart#L106-L120)